naomi-setup.sh

```diff
@@ -7,54 +7,239 @@
 # install naomi & requirements in their
 # respective directories.
 #########################################
+OPTION="0"
+for var in "$@"; do
+    if [ "$var" = "--virtualenv" ]; then
+        OPTION="1"
+    fi
+    if [ "$var" = "--local" ]; then
+        OPTION="2"
+    fi
+    if [ "$var" = "--primary" ]; then
+        OPTION="3"
+    fi
+    if [ "$var" = "--help" ]; then
+        echo "USAGE: $0 [--virtualenv | --local | --primary | --help]"
+        echo
+        echo "  --virtualenv - install Naomi using a virtualenv environment for Naomi"
+        echo "                 (this is the recommended choice. You will need to issue"
+        echo "                 'workon Naomi' before installing additional libraries"
+        echo "                 for Naomi)"
+        echo
+        echo "  --local      - download, compile and install a special copy of Python 3"
+        echo "                 for Naomi (does not work for all distros)"
+        echo
+        echo "  --primary    - use your primary Python 3 environment"
+        echo "                 (this can be dangerous, it can lead to a broken python"
+        echo "                 environment on your system, which other software may"
+        echo "                 depend on. This is the simplest setup, but only recommended"
+        echo "                 when Naomi is the primary program you intend to run on this"
+        echo "                 computer. You will probably need to use the 'pip3' command"
+        echo "                 to install additional libraries for Naomi.)"
+        echo "                 USE AT YOUR OWN RISK!"
+        echo
+        echo "  --help       - Print this message and exit"
+        exit 0
+    fi
+done
+if [ $OPTION = "0" ]; then
+    echo 'There are three methods to install Naomi:'
+    echo
+    echo '1) Use virtualenvwrapper to create a virtual Python 3 environment for Naomi (recommended)'
+    echo '2) Download, compile, and install a local copy of Python for Naomi (may not work for some users)'
+    echo '3) Use your primary Python 3 environment (this can be dangerous and can break other software on your system. It is only recommended for machines you intend to devote to running Naomi, like a virtual machine or Raspberry Pi)'
```

Comment:
Somewhere in this, we should add a comment about the fact this script will try sudoing without a password. You will not see line 68. This gets skipped on a RPI image because it just starts installing for you after choosing an option. It just depends on how the sudo powers are setup. Some distros will always ask for a password. Some will remember during a session, and so on.

Reply:
The program now stops and waits for input on any use of sudo with a red "Notice" marker and a green copy of the exact line to be executed. The user can press "q" to exit without proceeding.

Reply:
Very nice, look forward to test it.